server/src/main/java/org/elasticsearch/action/admin/cluster/stats/ClusterStatsResponse.java

```diff
@@ -150,4 +162,59 @@ public String toString() {
         return Strings.toString(this, true, true);
     }
 
+    public record RemoteClusterStats(
+        String clusterUUID,
+        String mode,
+        boolean skipUnavailable,
+        String transportCompress,
+        Set<String> versions,
+        String status,
+        long nodesCount,
+        long shardsCount,
+        long indicesCount,
+        long indicesBytes,
+        long heapBytes,
+        long memBytes
+    ) implements ToXContentFragment {
+        public RemoteClusterStats(
+            RemoteClusterStatsResponse remoteResponse,
+            String mode,
+            boolean skipUnavailable,
+            String transportCompress
+        ) {
+            this(
+                remoteResponse == null ? "unavailable" : remoteResponse.getClusterUUID(),
```

Comment:
I'd rather we didn't handle `null` in this contructor. Let's just make sure we don't pass in a `null` even in the failure case.

Reply:
The null here means the remote was unavailable. If we make `RemoteClusterStats` mutable it may not be necessary.

Reply:
Yes, but I'd prefer us to convert the error case into a suitable non-null `remoteResponse` in the caller rather than having all these null checks here. It could be a static constant with the fields already filled in.